lib/tasks/whitehall_importer.rb

```diff
@@ -0,0 +1,121 @@
+# frozen_string_literal: true
+
+module Tasks
+  class WhitehallImporter
+    SUPPORTED_WHITEHALL_STATES = %w(draft published rejected submitted superseded).freeze
+    SUPPORTED_LOCALES = %w(en).freeze
+
+    def initialize(whitehall_document_id, whitehall_document)
+      @whitehall_document_id = whitehall_document_id
+      @whitehall_document = whitehall_document
+      @whitehall_import = store_json_blob
+    end
+
+    def import
+      ActiveRecord::Base.transaction do
+        document = create_or_update_document
+
+        @whitehall_document["editions"].each_with_index do |edition, edition_number|
```

Comment:
It'd be more idiomatic with GOV.UK ruby to set this (and the other instance variables) with an attr_reader and not use the `@` to access. For example: https://github.com/alphagov/content-publisher/blob/9e3c55538077b375341f053aa350ca7f2e4ea486/lib/govspeak_document.rb#L4-L9

Reply:
Thanks for the example.  I've updated this PR now.